### ISSUE TITLE
how to build docker image for raspberry pi

### ISSUE THREAD

**Issue by marija185** [Engineering-Research-and-Development/iotagent-opcua#44]
Hello @gabrieledeluca 
we are using the docker image iotagent4fiware/iotagent-opcua:v1.3.7 successfully on a PC laptop to connect to OPC UA Omron PLC (btw the latest image does not work). We run orion and mongo on a remote server. Now, we wanted to replace the PC laptop with Raspberry pi and we have the following issue:
```
iotage_1  | standard_init_linux.go:219: exec user process caused: exec format error
```
I found the explanation here:
<https://qiita.com/reireias/items/03bfdaeb7f3afd7c4339>
So the solution is to build the iotagent-opcua image for ARM linux.
Can you please help me with this step? Can I use the dockerbuild file in this repository to do it myself? And how to make the right version that will provide a compatible version like v1.3.7 tag that currently works.

Thank you!
Best regards

Marija

**Comment by marija185** [comment]
Hello!
I solved it and it works now on Raspberry pi.
I followed your docs in the docker folder but I needed to do something first:

In the cloned repository, I needed to switch to tag v1.3.7 first. Then, modify your Dockerfile by changing node to arm32v7/node, line 4:

```
FROM arm32v7/node:${NODE_VERSION}-stretch
```

And execute the following command:
```
docker build -t iot-agent . --no-cache --build-arg DOWNLOAD=latest
```
Because this takes the current branch of the repository on the local disk.

That is it!
bye

**Comment by marija185** [comment]
Hello again!
I have a problem with the data flow. I tried to play with polling parameters but the behavior is always the same: I got data only for one second and I can see the entities on OCB, but after that, I got the message:
```
iotage_1  | time=2021-03-12T07:50:37.431Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.CommandService | msg=Executed expiration daemon
```
and data stop coming. When I restart the docker the data are again fetched for the first second and then nothing.

I also tried to build the docker image from the master branch and the behavior is the same. Is it the problem that raspberry pi has only 4GB of RAM? Can we slow down the data somehow to have new values every 2 seconds for example?

We need to present only 4 sensors to OCB.

Thank you and best regards

Marija
